### PR TITLE
Add website/tests/node_modules and settings.json to .gitignore

### PR DESCRIPTION
**Pull request by SPDonaghy**
### Description

Added src/website/tests/node_modules and settings.json to the .gitignore file to ensure these are not tracked by git.

### Verification

CI passes

### Resources

Slack Reference:
https://ubcsailbot.slack.com/archives/C042XBRFBKJ/p1716044689558669

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -12,6 +12,7 @@
 
 # microsoft c/c++ intellisense (disabled in favor of clangd)
 /.vscode/c_cpp_properties.json
+/.vscode/settings.json
```
Comment (patrick-5546) [inline .gitignore:15]: Is settings.json still being autogenerated? What is its contents? Haven't had this happen to me before

Comment (SPDonaghy) [reply .gitignore:15]: Yeah I just did a fresh rebuild of everything and it showed up for me. I can check the contents when I get home, I didn't look too closely at what was in there

Comment (patrick-5546) [reply .gitignore:15]: it doesn't for me, maybe due to a difference in our personal settings

Comment (SPDonaghy) [reply .gitignore:15]: That's probably what it is, do you think settings.json should be tracked?

Comment (patrick-5546) [reply .gitignore:15]: maybe just ignore the entire directory since we don't use it. All settings should be in the code-workspace file

Comment (SPDonaghy) [reply .gitignore:15]: Yeah thats a good point, I'll do that

File: .gitignore [hunk 2]
```diff
@@ -32,3 +33,6 @@ venv/
 .cache/
 # Build
 site/
+
+# web
+/src/website/tests/node_modules/*
```
Comment (JordanChen123) [inline .gitignore:38]: Looks good to me!

**Review by JordanChen123 — APPROVED** [review]

**Pull request merged**